app/src/main/java/com/example/tasky/login/components/TaskyTextField.kt

```diff
@@ -0,0 +1,145 @@
+package com.example.tasky.login.components
+
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Visibility
+import androidx.compose.material.icons.filled.VisibilityOff
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.TextFieldDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.text.input.TextFieldValue
+import androidx.compose.ui.text.input.VisualTransformation
+import androidx.compose.ui.unit.dp
+import com.example.tasky.R
+import com.example.tasky.ui.theme.DarkGray
+import com.example.tasky.ui.theme.Gray
+import com.example.tasky.ui.theme.Gray2
+import com.example.tasky.ui.theme.Green
+import com.example.tasky.ui.theme.LightBlue2
+import com.example.tasky.ui.theme.LightGray2
+import com.example.tasky.ui.theme.Red
+
+@Composable
+fun TaskyTextField(
+    modifier: Modifier = Modifier,
```

Comment:
Modifier should be the first optional parameter. And optional parameters should happen after the mandatory parameters

In this case, value and onValueChange are the only mandatory parameter since they dont have a default value


So value should appear before Modifier (i understand if onValueChange you leave it at the end, to use a high order function)


This is, because if not, to create a TaskyTextField, you need to always use named parameters just to set a value